pgrx/src/spi/client.rs

```diff
@@ -5,22 +5,26 @@ use std::ptr::NonNull;
 use crate::pg_sys::{self, PgOid};
 use crate::spi::{PreparedStatement, Query, Spi, SpiCursor, SpiError, SpiResult, SpiTupleTable};
 
-// TODO: should `'conn` be invariant?
-pub struct SpiClient<'conn> {
-    __marker: PhantomData<&'conn SpiConnection>,
-}
+pub struct SpiClient;
```

Comment:
This allows creating an SpiClient out of nothing and then forcing `SPI_finish` prematurely, simply by writing
```rust
let x = SpiClient;
mem::drop(x);
```

This could then cause problems where memory is now freed by Postgres but the Rust lifetime may not have ended. This approach will not work.

Reply:
oh yeah, that shouldn't have been `pub`.  `pub(super)` is what I wanted!

Reply:
`Spi::connect` depends on exposing access to `SpiClient` in its current API.

Reply:
yeah yeah yeah.  fix incoming....

Reply:
smash that refresh button!